### ISSUE TITLE
AttributeError: 'Uploader' object has no attribute 'file_size'

### ISSUE THREAD

**Issue by isears** [galaxyproject/galaxy-upload#1]
Hi,

I get the above error when trying to upload to https://usegalaxy.org

Any ideas?

command:
```
export GALAXY_API_KEY=XXXXXXXXXXX
galaxy-upload --url https://usegalaxy.org --history-name Upload R1.fastq.gz R2.fastq.gz
```

Full stack trace:
```
Traceback (most recent call last):
  File "/users/isears1/anaconda/rnaseq/bin/galaxy-upload", line 8, in <module>
    sys.exit(main())
  File "/users/isears1/.local/lib/python3.7/site-packages/click/core.py", line 1130, in __call__
    return self.main(*args, **kwargs)
  File "/users/isears1/.local/lib/python3.7/site-packages/click/core.py", line 1055, in main
    rv = self.invoke(ctx)
  File "/users/isears1/.local/lib/python3.7/site-packages/click/core.py", line 1404, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/users/isears1/.local/lib/python3.7/site-packages/click/core.py", line 760, in invoke
    return __callback(*args, **kwargs)
  File "/users/isears1/anaconda/rnaseq/lib/python3.7/site-packages/galaxy_upload/upload.py", line 186, in main
    upload_file(url, path, api_key, **kwargs)
  File "/users/isears1/anaconda/rnaseq/lib/python3.7/site-packages/galaxy_upload/upload.py", line 100, in upload_file
    bar, task_id = make_bar(file_name, total=uploader.file_size)
AttributeError: 'Uploader' object has no attribute 'file_size'
```

Environment information:
```
python --version
Python 3.7.7

conda --version
conda 4.5.4
```

Install information:
```
python -m pip install galaxy-upload
Collecting galaxy-upload
  Using cached galaxy_upload-0.0.1-py3-none-any.whl (7.9 kB)
Requirement already satisfied: click-option-group in /gpfs/home/isears1/anaconda/rnaseq/lib/python3.7/site-packages (from galaxy-upload) (0.5.3)
Requirement already satisfied: click in /gpfs/home/isears1/.local/lib/python3.7/site-packages (from galaxy-upload) (8.1.3)
Requirement already satisfied: tuspy in /gpfs/home/isears1/anaconda/rnaseq/lib/python3.7/site-packages (from galaxy-upload) (1.0.0)
Requirement already satisfied: rich in /gpfs/home/isears1/.local/lib/python3.7/site-packages (from galaxy-upload) (12.4.4)
Requirement already satisfied: bioblend>=0.18.0 in /gpfs/home/isears1/anaconda/rnaseq/lib/python3.7/site-packages (from galaxy-upload) (0.18.0)
Requirement already satisfied: requests>=2.20.0 in /gpfs/home/isears1/.local/lib/python3.7/site-packages (from bioblend>=0.18.0->galaxy-upload) (2.28.1)
Requirement already satisfied: boto>=2.9.7 in /gpfs/home/isears1/anaconda/rnaseq/lib/python3.7/site-packages (from bioblend>=0.18.0->galaxy-upload) (2.49.0)
Requirement already satisfied: requests-toolbelt!=0.9.0,>=0.5.1 in /gpfs/home/isears1/anaconda/rnaseq/lib/python3.7/site-packages (from bioblend>=0.18.0->galaxy-upload) (0.9.1)
Requirement already satisfied: pyyaml in /gpfs/home/isears1/.local/lib/python3.7/site-packages (from bioblend>=0.18.0->galaxy-upload) (6.0)
Requirement already satisfied: idna<4,>=2.5 in /gpfs/home/isears1/.local/lib/python3.7/site-packages (from requests>=2.20.0->bioblend>=0.18.0->galaxy-upload) (3.3)
Requirement already satisfied: urllib3<1.27,>=1.21.1 in /gpfs/home/isears1/.local/lib/python3.7/site-packages (from requests>=2.20.0->bioblend>=0.18.0->galaxy-upload) (1.26.9)
Requirement already satisfied: charset-normalizer<3,>=2 in /gpfs/home/isears1/.local/lib/python3.7/site-packages (from requests>=2.20.0->bioblend>=0.18.0->galaxy-upload) (2.1.0)
Requirement already satisfied: certifi>=2017.4.17 in /gpfs/home/isears1/.local/lib/python3.7/site-packages (from requests>=2.20.0->bioblend>=0.18.0->galaxy-upload) (2022.6.15)
Requirement already satisfied: importlib-metadata in /gpfs/home/isears1/.local/lib/python3.7/site-packages (from click->galaxy-upload) (4.12.0)
Requirement already satisfied: typing-extensions>=3.6.4 in /gpfs/home/isears1/.local/lib/python3.7/site-packages (from importlib-metadata->click->galaxy-upload) (4.2.0)
Requirement already satisfied: zipp>=0.5 in /gpfs/home/isears1/.local/lib/python3.7/site-packages (from importlib-metadata->click->galaxy-upload) (3.8.0)
Requirement already satisfied: commonmark<0.10.0,>=0.9.0 in /gpfs/home/isears1/.local/lib/python3.7/site-packages (from rich->galaxy-upload) (0.9.1)
Requirement already satisfied: pygments<3.0.0,>=2.6.0 in /gpfs/home/isears1/anaconda/rnaseq/lib/python3.7/site-packages (from rich->galaxy-upload) (2.11.2)
Requirement already satisfied: aiohttp>=3.6.2 in /gpfs/home/isears1/anaconda/rnaseq/lib/python3.7/site-packages (from tuspy->galaxy-upload) (3.8.1)
Requirement already satisfied: future>=0.16.0 in /gpfs/home/isears1/.local/lib/python3.7/site-packages (from tuspy->galaxy-upload) (0.18.2)
Requirement already satisfied: tinydb>=3.5.0 in /gpfs/home/isears1/anaconda/rnaseq/lib/python3.7/site-packages (from tuspy->galaxy-upload) (4.7.0)
Requirement already satisfied: six>=1.11.0 in /gpfs/home/isears1/.local/lib/python3.7/site-packages (from tuspy->galaxy-upload) (1.16.0)
Requirement already satisfied: aiosignal>=1.1.2 in /gpfs/home/isears1/anaconda/rnaseq/lib/python3.7/site-packages (from aiohttp>=3.6.2->tuspy->galaxy-upload) (1.2.0)
Requirement already satisfied: frozenlist>=1.1.1 in /gpfs/home/isears1/anaconda/rnaseq/lib/python3.7/site-packages (from aiohttp>=3.6.2->tuspy->galaxy-upload) (1.3.0)
Requirement already satisfied: attrs>=17.3.0 in /gpfs/home/isears1/.local/lib/python3.7/site-packages (from aiohttp>=3.6.2->tuspy->galaxy-upload) (21.4.0)
Requirement already satisfied: yarl<2.0,>=1.0 in /gpfs/home/isears1/anaconda/rnaseq/lib/python3.7/site-packages (from aiohttp>=3.6.2->tuspy->galaxy-upload) (1.7.2)
Requirement already satisfied: asynctest==0.13.0 in /gpfs/home/isears1/anaconda/rnaseq/lib/python3.7/site-packages (from aiohttp>=3.6.2->tuspy->galaxy-upload) (0.13.0)
Requirement already satisfied: async-timeout<5.0,>=4.0.0a3 in /gpfs/home/isears1/anaconda/rnaseq/lib/python3.7/site-packages (from aiohttp>=3.6.2->tuspy->galaxy-upload) (4.0.2)
Requirement already satisfied: multidict<7.0,>=4.5 in /gpfs/home/isears1/anaconda/rnaseq/lib/python3.7/site-packages (from aiohttp>=3.6.2->tuspy->galaxy-upload) (6.0.2)
Installing collected packages: galaxy-upload
Successfully installed galaxy-upload-0.0.1
```

System: RHEL 7.7

**Comment by jonahanderson** [comment]
having the same issue ^

**Comment by jonahanderson** [comment]
I fixed the issue but I cannot push my branch or make a pull request. Basically just replace the upload.py file with this:

`#!/usr/bin/env python3
"""Upload files to a Galaxy instance."""
import os
import sys
import traceback

import click
import rich.console
import rich.progress
from bioblend import ConnectionError
from bioblend.galaxy import GalaxyInstance
from click_option_group import RequiredMutuallyExclusiveOptionGroup, optgroup
from tusclient.fingerprint import fingerprint

from .history import get_histories, make_table


def make_bar(file_name, total=None):
    columns = (
        rich.progress.TextColumn("[progress.description]{task.description}"),
        rich.progress.BarColumn(),
        rich.progress.DownloadColumn(),
        rich.progress.TransferSpeedColumn(),
        rich.progress.TextColumn("eta"),
        rich.progress.TimeRemainingColumn(),
    )

    bar = rich.progress.Progress(*columns)
    task_id = bar.add_task(file_name, total=total)
    return (bar, task_id)


def find_history(gi, history_name, ignore_case):
    histories = get_histories(gi, ignore_case=ignore_case, name=history_name)
    if not histories:
        message = f"ERROR: No histories matching {history_name} found!"
        click.echo(click.style(message, bold=True, fg="red"), err=True)
        sys.exit(1)
    elif len(histories) > 1:
        table = make_table(quiet=False, histories=histories)
        console = rich.console.Console()
        console.print(table)
        message = f"ERROR: Multiple histories matching {history_name} found! Use --history-id to select one."
        click.echo(click.style(message, bold=True, fg="red"), err=True)
        sys.exit(1)
    return histories[0]


def upload_file(
    url,
    path,
    api_key,
    history_id=None,
    history_name=None,
    ignore_case=False,
    file_type="auto",
    dbkey="?",
    space_to_tab=False,
    auto_decompress=False,
    file_name=None,
    storage=None,
    silent=False,
    debug=False,
):
    file_name = file_name or os.path.basename(path)

    gi = GalaxyInstance(url, api_key)

    upload_kwargs = {
        "file_type": file_type,
        "dbkey": dbkey,
        "file_name": file_name,
        "space_to_tab": space_to_tab,
        "auto_decompress": auto_decompress,
    }

    if not history_id:
        history = find_history(gi, history_name, ignore_case)
        history_id = history["id"]
        if debug:
            click.echo(
                f"History name search '{history_name}' resolved to history {history_id}: {history['name']}"
            )

    try:
        if silent:
            gi.tools.upload_file_tus(
                path,
                history_id,
                storage=storage,
                **upload_kwargs,
            )
            return

        uploader = gi.get_tus_uploader(
            path,
            storage=storage,
        )
        
        bar, task_id = make_bar(file_name, total=uploader.get_file_size())

        with bar:
            last_offset = 0
            while uploader.offset < uploader.get_file_size():
                uploader.upload_chunk()
                bar.update(task_id, advance=(uploader.offset - last_offset))
                last_offset = uploader.offset

        gi.tools.post_to_fetch(path, history_id, uploader.session_id, **upload_kwargs)

    except ConnectionError as exc:
        if exc.status_code == 404 and storage:
            fingerprinter = fingerprint.Fingerprint()
            with open(path, "rb") as fh:
                fp_hash = fingerprinter.get_fingerprint(fh)
            message = (
                f"Unable to resume, previous upload may have been removed from server (hint: remove {fp_hash}"
                f" from {storage} or change storage to reupload from the start: {exc}"
            )
        else:
            message = str(exc)
        message = f"ERROR: {message}"
        if debug:
            click.echo(traceback.format_exc(), nl=False)
        click.echo(click.style(message, bold=True, fg="red"), err=True)


@click.command()
@optgroup.group("History Selection Options", cls=RequiredMutuallyExclusiveOptionGroup)
@optgroup.option("--history-id", type=str, help="Target history ID")
@optgroup.option("--history-name", type=str, help="Target history name")
@optgroup.group("History Options")
@optgroup.option(
    "--ignore-case", "-i", is_flag=True, help="Ignore case when matching history names"
)
@optgroup.group("Galaxy Server Options")
@optgroup.option(
    "--url",
    envvar="GALAXY_URL",
    default="http://localhost:8080",
    help="URL of Galaxy instance",
)
@optgroup.option(
    "--api-key",
    envvar="GALAXY_API_KEY",
    required=True,
    help="API key for Galaxy instance",
)
@optgroup.group("Upload Options")
@optgroup.option(
    "--file-type", default="auto", type=str, help="Galaxy file type to use"
)
@optgroup.option("--dbkey", default="?", type=str, help="Genome Build for dataset")
@optgroup.option(
    "--space-to-tab/--no-space-to-tab", default=False, help="Convert spaces to tabs"
)
@optgroup.option(
    "--auto-decompress/--no-auto-decompress",
    default=True,
    help="Automatically decompress after upload",
)
@optgroup.option(
    "--file-name",
    type=str,
    help="Filename to use in Galaxy history, if different from path",
)
@optgroup.group("General Options")
@optgroup.option(
    "--storage", type=click.Path(), required=False, help="Store URLs to resume here"
)
@optgroup.option(
    "--silent", is_flag=True, default=False, help="No output while uploading"
)
@optgroup.option("--debug", is_flag=True, default=False, help="Debug output")
@click.argument("path", type=click.Path(), nargs=-1)
def main(url, path, api_key, **kwargs):
    paths = path
    if not paths:
        message = "WARNING: No paths to upload specified (see --help)"
        click.echo(click.style(message, bold=True, fg="yellow"), err=True)
    elif len(paths) > 1 and kwargs["file_name"]:
        message = "ERROR: --file-name option cannot be used with multiple paths"
        click.echo(click.style(message, bold=True, fg="red"), err=True)
        sys.exit(1)
    for path in paths:
        upload_file(url, path, api_key, **kwargs)


if __name__ == "__main__":
    main()
`

**Comment by jonahanderson** [comment]
> Hi,
> 
> I get the above error when trying to upload to https://usegalaxy.org
> 
> Any ideas?
> 
> command:
> 
> ```
> export GALAXY_API_KEY=XXXXXXXXXXX
> galaxy-upload --url https://usegalaxy.org --history-name Upload R1.fastq.gz R2.fastq.gz
> ```
> 
> Full stack trace:
> 
> ```
> Traceback (most recent call last):
>   File "/users/isears1/anaconda/rnaseq/bin/galaxy-upload", line 8, in <module>
>     sys.exit(main())
>   File "/users/isears1/.local/lib/python3.7/site-packages/click/core.py", line 1130, in __call__
>     return self.main(*args, **kwargs)
>   File "/users/isears1/.local/lib/python3.7/site-packages/click/core.py", line 1055, in main
>     rv = self.invoke(ctx)
>   File "/users/isears1/.local/lib/python3.7/site-packages/click/core.py", line 1404, in invoke
>     return ctx.invoke(self.callback, **ctx.params)
>   File "/users/isears1/.local/lib/python3.7/site-packages/click/core.py", line 760, in invoke
>     return __callback(*args, **kwargs)
>   File "/users/isears1/anaconda/rnaseq/lib/python3.7/site-packages/galaxy_upload/upload.py", line 186, in main
>     upload_file(url, path, api_key, **kwargs)
>   File "/users/isears1/anaconda/rnaseq/lib/python3.7/site-packages/galaxy_upload/upload.py", line 100, in upload_file
>     bar, task_id = make_bar(file_name, total=uploader.file_size)
> AttributeError: 'Uploader' object has no attribute 'file_size'
> ```
> 
> Environment information:
> 
> ```
> python --version
> Python 3.7.7
> 
> conda --version
> conda 4.5.4
> ```
> 
> Install information:
> 
> ```
> python -m pip install galaxy-upload
> Collecting galaxy-upload
>   Using cached galaxy_upload-0.0.1-py3-none-any.whl (7.9 kB)
> Requirement already satisfied: click-option-group in /gpfs/home/isears1/anaconda/rnaseq/lib/python3.7/site-packages (from galaxy-upload) (0.5.3)
> Requirement already satisfied: click in /gpfs/home/isears1/.local/lib/python3.7/site-packages (from galaxy-upload) (8.1.3)
> Requirement already satisfied: tuspy in /gpfs/home/isears1/anaconda/rnaseq/lib/python3.7/site-packages (from galaxy-upload) (1.0.0)
> Requirement already satisfied: rich in /gpfs/home/isears1/.local/lib/python3.7/site-packages (from galaxy-upload) (12.4.4)
> Requirement already satisfied: bioblend>=0.18.0 in /gpfs/home/isears1/anaconda/rnaseq/lib/python3.7/site-packages (from galaxy-upload) (0.18.0)
> Requirement already satisfied: requests>=2.20.0 in /gpfs/home/isears1/.local/lib/python3.7/site-packages (from bioblend>=0.18.0->galaxy-upload) (2.28.1)
> Requirement already satisfied: boto>=2.9.7 in /gpfs/home/isears1/anaconda/rnaseq/lib/python3.7/site-packages (from bioblend>=0.18.0->galaxy-upload) (2.49.0)
> Requirement already satisfied: requests-toolbelt!=0.9.0,>=0.5.1 in /gpfs/home/isears1/anaconda/rnaseq/lib/python3.7/site-packages (from bioblend>=0.18.0->galaxy-upload) (0.9.1)
> Requirement already satisfied: pyyaml in /gpfs/home/isears1/.local/lib/python3.7/site-packages (from bioblend>=0.18.0->galaxy-upload) (6.0)
> Requirement already satisfied: idna<4,>=2.5 in /gpfs/home/isears1/.local/lib/python3.7/site-packages (from requests>=2.20.0->bioblend>=0.18.0->galaxy-upload) (3.3)
> Requirement already satisfied: urllib3<1.27,>=1.21.1 in /gpfs/home/isears1/.local/lib/python3.7/site-packages (from requests>=2.20.0->bioblend>=0.18.0->galaxy-upload) (1.26.9)
> Requirement already satisfied: charset-normalizer<3,>=2 in /gpfs/home/isears1/.local/lib/python3.7/site-packages (from requests>=2.20.0->bioblend>=0.18.0->galaxy-upload) (2.1.0)
> Requirement already satisfied: certifi>=2017.4.17 in /gpfs/home/isears1/.local/lib/python3.7/site-packages (from requests>=2.20.0->bioblend>=0.18.0->galaxy-upload) (2022.6.15)
> Requirement already satisfied: importlib-metadata in /gpfs/home/isears1/.local/lib/python3.7/site-packages (from click->galaxy-upload) (4.12.0)
> Requirement already satisfied: typing-extensions>=3.6.4 in /gpfs/home/isears1/.local/lib/python3.7/site-packages (from importlib-metadata->click->galaxy-upload) (4.2.0)
> Requirement already satisfied: zipp>=0.5 in /gpfs/home/isears1/.local/lib/python3.7/site-packages (from importlib-metadata->click->galaxy-upload) (3.8.0)
> Requirement already satisfied: commonmark<0.10.0,>=0.9.0 in /gpfs/home/isears1/.local/lib/python3.7/site-packages (from rich->galaxy-upload) (0.9.1)
> Requirement already satisfied: pygments<3.0.0,>=2.6.0 in /gpfs/home/isears1/anaconda/rnaseq/lib/python3.7/site-packages (from rich->galaxy-upload) (2.11.2)
> Requirement already satisfied: aiohttp>=3.6.2 in /gpfs/home/isears1/anaconda/rnaseq/lib/python3.7/site-packages (from tuspy->galaxy-upload) (3.8.1)
> Requirement already satisfied: future>=0.16.0 in /gpfs/home/isears1/.local/lib/python3.7/site-packages (from tuspy->galaxy-upload) (0.18.2)
> Requirement already satisfied: tinydb>=3.5.0 in /gpfs/home/isears1/anaconda/rnaseq/lib/python3.7/site-packages (from tuspy->galaxy-upload) (4.7.0)
> Requirement already satisfied: six>=1.11.0 in /gpfs/home/isears1/.local/lib/python3.7/site-packages (from tuspy->galaxy-upload) (1.16.0)
> Requirement already satisfied: aiosignal>=1.1.2 in /gpfs/home/isears1/anaconda/rnaseq/lib/python3.7/site-packages (from aiohttp>=3.6.2->tuspy->galaxy-upload) (1.2.0)
> Requirement already satisfied: frozenlist>=1.1.1 in /gpfs/home/isears1/anaconda/rnaseq/lib/python3.7/site-packages (from aiohttp>=3.6.2->tuspy->galaxy-upload) (1.3.0)
> Requirement already satisfied: attrs>=17.3.0 in /gpfs/home/isears1/.local/lib/python3.7/site-packages (from aiohttp>=3.6.2->tuspy->galaxy-upload) (21.4.0)
> Requirement already satisfied: yarl<2.0,>=1.0 in /gpfs/home/isears1/anaconda/rnaseq/lib/python3.7/site-packages (from aiohttp>=3.6.2->tuspy->galaxy-upload) (1.7.2)
> Requirement already satisfied: asynctest==0.13.0 in /gpfs/home/isears1/anaconda/rnaseq/lib/python3.7/site-packages (from aiohttp>=3.6.2->tuspy->galaxy-upload) (0.13.0)
> Requirement already satisfied: async-timeout<5.0,>=4.0.0a3 in /gpfs/home/isears1/anaconda/rnaseq/lib/python3.7/site-packages (from aiohttp>=3.6.2->tuspy->galaxy-upload) (4.0.2)
> Requirement already satisfied: multidict<7.0,>=4.5 in /gpfs/home/isears1/anaconda/rnaseq/lib/python3.7/site-packages (from aiohttp>=3.6.2->tuspy->galaxy-upload) (6.0.2)
> Installing collected packages: galaxy-upload
> Successfully installed galaxy-upload-0.0.1
> ```
> 
> System: RHEL 7.7

I fixed the issue but I cannot push my branch or create a pull request. Basically just replace the upload.py file with this:

`#!/usr/bin/env python3
"""Upload files to a Galaxy instance."""
import os
import sys
import traceback

import click
import rich.console
import rich.progress
from bioblend import ConnectionError
from bioblend.galaxy import GalaxyInstance
from click_option_group import RequiredMutuallyExclusiveOptionGroup, optgroup
from tusclient.fingerprint import fingerprint

from .history import get_histories, make_table


def make_bar(file_name, total=None):
    columns = (
        rich.progress.TextColumn("[progress.description]{task.description}"),
        rich.progress.BarColumn(),
        rich.progress.DownloadColumn(),
        rich.progress.TransferSpeedColumn(),
        rich.progress.TextColumn("eta"),
        rich.progress.TimeRemainingColumn(),
    )

    bar = rich.progress.Progress(*columns)
    task_id = bar.add_task(file_name, total=total)
    return (bar, task_id)


def find_history(gi, history_name, ignore_case):
    histories = get_histories(gi, ignore_case=ignore_case, name=history_name)
    if not histories:
        message = f"ERROR: No histories matching {history_name} found!"
        click.echo(click.style(message, bold=True, fg="red"), err=True)
        sys.exit(1)
    elif len(histories) > 1:
        table = make_table(quiet=False, histories=histories)
        console = rich.console.Console()
        console.print(table)
        message = f"ERROR: Multiple histories matching {history_name} found! Use --history-id to select one."
        click.echo(click.style(message, bold=True, fg="red"), err=True)
        sys.exit(1)
    return histories[0]


def upload_file(
    url,
    path,
    api_key,
    history_id=None,
    history_name=None,
    ignore_case=False,
    file_type="auto",
    dbkey="?",
    space_to_tab=False,
    auto_decompress=False,
    file_name=None,
    storage=None,
    silent=False,
    debug=False,
):
    file_name = file_name or os.path.basename(path)

    gi = GalaxyInstance(url, api_key)

    upload_kwargs = {
        "file_type": file_type,
        "dbkey": dbkey,
        "file_name": file_name,
        "space_to_tab": space_to_tab,
        "auto_decompress": auto_decompress,
    }

    if not history_id:
        history = find_history(gi, history_name, ignore_case)
        history_id = history["id"]
        if debug:
            click.echo(
                f"History name search '{history_name}' resolved to history {history_id}: {history['name']}"
            )

    try:
        if silent:
            gi.tools.upload_file_tus(
                path,
                history_id,
                storage=storage,
                **upload_kwargs,
            )
            return

        uploader = gi.get_tus_uploader(
            path,
            storage=storage,
        )
        
        bar, task_id = make_bar(file_name, total=uploader.get_file_size())

        with bar:
            last_offset = 0
            while uploader.offset < uploader.get_file_size():
                uploader.upload_chunk()
                bar.update(task_id, advance=(uploader.offset - last_offset))
                last_offset = uploader.offset

        gi.tools.post_to_fetch(path, history_id, uploader.session_id, **upload_kwargs)

    except ConnectionError as exc:
        if exc.status_code == 404 and storage:
            fingerprinter = fingerprint.Fingerprint()
            with open(path, "rb") as fh:
                fp_hash = fingerprinter.get_fingerprint(fh)
            message = (
                f"Unable to resume, previous upload may have been removed from server (hint: remove {fp_hash}"
                f" from {storage} or change storage to reupload from the start: {exc}"
            )
        else:
            message = str(exc)
        message = f"ERROR: {message}"
        if debug:
            click.echo(traceback.format_exc(), nl=False)
        click.echo(click.style(message, bold=True, fg="red"), err=True)


@click.command()
@optgroup.group("History Selection Options", cls=RequiredMutuallyExclusiveOptionGroup)
@optgroup.option("--history-id", type=str, help="Target history ID")
@optgroup.option("--history-name", type=str, help="Target history name")
@optgroup.group("History Options")
@optgroup.option(
    "--ignore-case", "-i", is_flag=True, help="Ignore case when matching history names"
)
@optgroup.group("Galaxy Server Options")
@optgroup.option(
    "--url",
    envvar="GALAXY_URL",
    default="http://localhost:8080",
    help="URL of Galaxy instance",
)
@optgroup.option(
    "--api-key",
    envvar="GALAXY_API_KEY",
    required=True,
    help="API key for Galaxy instance",
)
@optgroup.group("Upload Options")
@optgroup.option(
    "--file-type", default="auto", type=str, help="Galaxy file type to use"
)
@optgroup.option("--dbkey", default="?", type=str, help="Genome Build for dataset")
@optgroup.option(
    "--space-to-tab/--no-space-to-tab", default=False, help="Convert spaces to tabs"
)
@optgroup.option(
    "--auto-decompress/--no-auto-decompress",
    default=True,
    help="Automatically decompress after upload",
)
@optgroup.option(
    "--file-name",
    type=str,
    help="Filename to use in Galaxy history, if different from path",
)
@optgroup.group("General Options")
@optgroup.option(
    "--storage", type=click.Path(), required=False, help="Store URLs to resume here"
)
@optgroup.option(
    "--silent", is_flag=True, default=False, help="No output while uploading"
)
@optgroup.option("--debug", is_flag=True, default=False, help="Debug output")
@click.argument("path", type=click.Path(), nargs=-1)
def main(url, path, api_key, **kwargs):
    paths = path
    if not paths:
        message = "WARNING: No paths to upload specified (see --help)"
        click.echo(click.style(message, bold=True, fg="yellow"), err=True)
    elif len(paths) > 1 and kwargs["file_name"]:
        message = "ERROR: --file-name option cannot be used with multiple paths"
        click.echo(click.style(message, bold=True, fg="red"), err=True)
        sys.exit(1)
    for path in paths:
        upload_file(url, path, api_key, **kwargs)


if __name__ == "__main__":
    main()
`

**Comment by jonahanderson** [comment]
@isears I fixed the issue but I cannot push my branch or create a pull request. @natefoo could you grant me access?

Basically just replace lines 100-107 in your upload.py file with this: 

```
        bar, task_id = make_bar(file_name, total=uploader.get_file_size())

        with bar:
            last_offset = 0
            while uploader.offset < uploader.get_file_size():
                uploader.upload_chunk()
                bar.update(task_id, advance=(uploader.offset - last_offset))
                last_offset = uploader.offset

```
                
The only change here is replacing the instances of `uploader.file_size` with `uploader.get_file_size()` - a change needed for tusclient 1.0.0.

Hope this helps!

**Comment by natefoo** [comment]
Sorry for not getting to this sooner, I was on vacation as tuspy 1.0.0 was released and just caught up to this now.

This should be fixed by #3,

**Comment by natefoo** [comment]
@jonahanderson you should be able to fork and PR from there, is this not working for you?
